server/actions/ssh/runScript.js

```diff
@@ -1,101 +1,111 @@
+/* eslint-disable prettier/prettier */
 /**
  * This file helps transfer the input.txt needed and then runs the script in the hidden directory
  */
 import { command, transfer } from './';
 import { saveStudents } from '../grade';
 
 // TODO: Need to split this file apart so that we make multiple queries
-export default function runScript(req) {
-  const { assignment, socketId } = req.body;
-  const { repo, input, path, bonusDate, name } = assignment;
-  const repoPath = `.private_repo/${path.match(/.*\/(.+?)\/?$/)[1]}`;
-
-  return command({
-    body: {
-      socketId,
-      command: `rm -r ${repoPath}`
+
+export default async function runScript(req) {
+    const { assignment, socketId } = req.body;
+    const { repo, input, path, bonusDate, name } = assignment;
+    const repoPath = `.private_repo/${path.match(/.*\/(.+?)\/?$/)[1]}`;
+
+    try {
+        await removeProjectDirFromPrivateRepoIfItExist(repoPath, socketId);
+        await transferFileContentToServer(socketId, input, `${repoPath}/input.txt`);
+        await copyFilesToGradeToPrivateRepo(repoPath, path, socketId);
+        const data = await getResultFromGradingScript(repoPath, bonusDate, socketId);
+        const previewList = await getHTMLOutPutFileList(name, socketId);
+
+        if (!data.graders || !previewList) {
+            return Promise.reject({ message: 'Error running script! Please make sure repository path is correct!' });
+        }
+
+        const bonusList = await getBonusList(name, socketId);
+        await saveStudentResults(data, previewList, repo, name, bonusList);
+
+        return data;
+    }catch(err) {
```

Comment:
I don't think you need this try catch here because you're catching it and just re-throwing the error

Reply:
Well the thing is that if I do not put the try and catch then if any of the await get rejected an exception will be thrown but the exception will not be "Promise.reject({ message: err });" So it wont show on the UI